### PR TITLE
Pass on 0.7

### PR DESCRIPTION
**Pull request by abelsiqueira**
- Deletes JuMP/MPB related code;
- Uses NLPModels 0.5.0;
- Use `Ref` on `ccall`s.

### CODE REVIEW

**Comment by dpo** [comment]
You're hitting a Homebrew.jl bug on macOS. What I do now locally is install CUTEst outside of Julia with `brew install cutest` and let BinDeps pick it up instead of trying to install it. You could try that in `.travis.yml`.

**Comment by abelsiqueira** [comment]
Let me try that.

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ branches:
 
 before_install:
   - if [ `uname` == "Linux" ]; then unset DY_LIBRARY_PATH; fi
-  - if [ `uname` == "Darwin" ]; then brew update && brew cask uninstall oclint && brew install gcc; fi
+  - if [ `uname` == "Darwin" ]; then brew update; brew tap optimizers/cutest; brew install cutest; brew install mastsif; for f in "archdefs" "sifdecode" "mastsif" "cutest"; do echo ". $(brew --prefix $f)/$f.bashrc" >> ~/.bashrc; done; fi
```
Comment (dpo) [inline .travis.yml]: You'd still need to install `gcc` I think. The hope is that Homebrew.jl wouldn't need to install anything (but you never know...)

Comment (dpo) [reply .travis.yml]: Oh and there's no need to cat the files into `.bashrc`, you could just source each of them.

**Comment by abelsiqueira** [comment]
Happy to say it passes. Thanks for the help.

File: .travis.yml [hunk 2]
```diff
@@ -19,16 +25,14 @@ branches:
 
 before_install:
   - if [ `uname` == "Linux" ]; then unset DY_LIBRARY_PATH; fi
-  - if [ `uname` == "Darwin" ]; then brew update && brew install gcc; fi
+  - if [ `uname` == "Darwin" ]; then brew update; brew cask uninstall oclint; brew install gcc; brew tap optimizers/cutest; brew install cutest; brew install mastsif; for f in "archdefs" "sifdecode" "mastsif" "cutest"; do source $(brew --prefix $f)/$f.bashrc >> ~/.bashrc; done; fi
```
Comment (dpo) [inline .travis.yml]: I don't think you need `>> ~/.bashrc` here.

File: .travis.yml [hunk 3]
```diff
@@ -19,16 +25,14 @@ branches:
...
-  - julia -E 'Pkg.clone(pwd());'
-  - julia -E 'Pkg.build("CUTEst");'
+  - julia -E 'VERSION >= v"0.7" && using Pkg; Pkg.clone(pwd()); Pkg.build("CUTEst");'
 
 script:
-  - julia --check-bounds=yes -E 'Pkg.test("CUTEst"; coverage=true)'
+  - julia --check-bounds=yes -E 'VERSION >= v"0.7" && using Pkg; Pkg.test("CUTEst"; coverage=true)'
 
 after_success:
   - julia -e 'Pkg.add("Documenter")'
```
Comment (dpo) [inline .travis.yml]: `using Pkg` if ≥ 0.7.

**Comment by dpo** [comment]
Why does Coveralls think coverage dropped to 0%?

**Comment by dpo** [comment]
I just enabled CodeCov for this repository. Here's the badge:
```
[![codecov](https://codecov.io/gh/JuliaSmoothOptimizers/CUTEst.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/JuliaSmoothOptimizers/CUTEst.jl)
```
In `.travis.yml`, you just need to change the last line to
```
julia -e 'VERSION >= v"0.7" && cd(Pkg.dir("CUTEst")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder()); Codecov.submit(process_folder())'
```

**Comment by dpo** [comment]
Excellent, thank you!

**Pull request merged**